lib/browser/api/web-contents.ts

```diff
@@ -134,37 +138,30 @@ WebContents.prototype.send = function (channel, ...args) {
   return this._send(false /* internal */, channel, args);
 };
 
-WebContents.prototype.postMessage = function (...args) {
-  if (Array.isArray(args[2])) {
-    args[2] = args[2].map(o => o instanceof MessagePortMain ? o._internalPort : o);
-  }
-  this._postMessage(...args);
-};
-
 WebContents.prototype._sendInternal = function (channel, ...args) {
   if (typeof channel !== 'string') {
     throw new Error('Missing required channel argument');
   }
 
   return this._send(true /* internal */, channel, args);
 };
-WebContents.prototype.sendToFrame = function (frame, channel, ...args) {
-  if (typeof channel !== 'string') {
-    throw new Error('Missing required channel argument');
-  } else if (!(typeof frame === 'number' || Array.isArray(frame))) {
-    throw new Error('Missing required frame argument (must be number or array)');
+
+function getWebFrame (contents: Electron.WebContents, frame: number | [number, number]) {
+  if (typeof frame === 'number') {
+    return webFrameMain.fromId(contents.mainFrame.processId, frame);
+  } else if (Array.isArray(frame) && frame.length === 2 && frame.every(value => typeof value === 'number')) {
+    return webFrameMain.fromId(frame[0], frame[1]);
+  } else {
+    throw new Error('Missing required frame argument (must be number or [processId, frameId])');
```

Comment:
We should guard against a `null` return value from `webFrameMain.fromId` here.

Reply:
I will do that, but need this #27068 to land first

Reply:
`WebContents.prototype.sendToFrame` and `WebContents.prototype._sendToFrameInternal` now checks the return value of `getWebFrame`